lesson_7/mosut_sorin/hw_lesson7_decorators_mosut_sorin2.py

```diff
@@ -0,0 +1,24 @@
+'''
+Given the following function:
+def divide(first_number, second_number):
+ return first_number / second_number
+Create a decorator called safe_divide that will output a message
+if the division cannot be performed, othervise it will return the result.
+'''
+
+
+def safe_divide(fnc):
+
+    def verif_fnc(*args):
+        if args[1] == 0:
+            return 'divide by 0'
+        else:
+            return f'{args[0] / args[1]:.2f}'
+
+    return verif_fnc
+
+@safe_divide
+def divide(first_number, second_number):
+    return first_number / second_number
+
+print(divide(20, 5))
```

Comment:
What would be the result if I try the following:
```python
a = divide(10, 2)
print(a * 5)
```
?

Reply:
The result of format() function is a string , so 
      a = divide(10, 2)
      print(a * 5)
=> 5.005.005.005.005.00
if we want to get a get a mathematical result from print(a * 5) we have to get a float from the divide() function.
we can do it like this:
         return float(f'{args[0] / args[1]:.2f}')

Reply:
Wouldn't be easier if we would not convert to string in the first place?

Reply:
I wanted to get a result with 2 numbers after the point , i think it looks better.
You will say that i could use the format only for printing , you would be right :).

Reply:
The main purpose of the function is to perform the division of two numbers and for such a function you would expect as result a number, not a string. How you choose to print the result afterwards it's a different matter.

Reply:
Should i make a commit to fix it ?